### ISSUE TITLE
Unreal Tournament (PAL) corrupts its own save files

### ISSUE THREAD

**Issue by Scr34mik** [libretro/flycast#994]
**_Platform / OS / Hardware:_**
Windows 10 20H2
8700K
RTX 2060
32Gb RAM
Retroarch and game is located on HDD

Github hash: 
ca2fa42

**_Description of the Issue_**
Since last update, Unreal Tournament (PAL) corrupts its own save files

**_Add a short, concise description of the issue here_**
Just playing one match in practice mode then closing-launching game is enough to get "save data corrupt" error

**_Debugging Steps Tested_**
* tried deleting UT save file via BIOS VMU manager
* tried using different (VMU1/Both/disabled) "per-content VMU" setting setups. Nothing helps.
* i was constantly deleting corrupted VMUs, and it didnt help.

**_Logs Gathered_**
- none

**_Screenshots_**
![image](https://user-images.githubusercontent.com/10966059/102956393-b8e04280-44e0-11eb-8ead-4b54100ac4e2.png)

**Comment by barbudreadmon** [comment]
It has probably been solved a long time ago, but there is currently an issue with the buildbot providing old builds

**Comment by taydan04** [comment]
@Scr34mik 

Yeah i had this issue. But I have a solution for you. Here is version aeb9e59 (Current commit) of the core that I built for windows x86_64 from source to get around it. Link will expire in a few days. Hmu if you don't get to it quick enough. Just unzip the dll file in to the cores folder within the RetroArch folder. :)

[flycast_libretro.zip](https://1drv.ms/u/s!Ap0VEg7UjYqTgo4SUL32AEm_x-IrtA?e=dG7pXg)

And if you don't trust the shared file (understandable) then there is a tutorial on the libretro site for building cores in windows [here](https://docs.libretro.com/development/retroarch/compilation/windows/).

**Comment by Scr34mik** [comment]
> 
> 
> @Scr34mik
> 
> Yeah i had this issue. But I have a solution for you. Here is version [aeb9e59](https://github.com/libretro/flycast/commit/aeb9e59beb572e856d0c0c199929d50a66d3f7b4) (Current commit) of the core that I built for windows x86_64 from source to get around it. Link will expire in a few days. Hmu if you don't get to it quick enough. Just unzip the dll file in to the cores folder within the RetroArch folder. :)
> 
> [flycast_libretro.zip](https://1drv.ms/u/s!Ap0VEg7UjYqTgo4SUL32AEm_x-IrtA?e=dG7pXg)
> 
> And if you don't trust the shared file (understandable) then there is a tutorial on the libretro site for building cores in windows [here](https://docs.libretro.com/development/retroarch/compilation/windows/).

Hi, thanks for the file, it'll be useful. 👍
I was in no rush however, and could just wait until buildbot bug is resolved.

UPD: Thanks. it seems issue with save fiels files is resolved. However, now there are corrupted polygons in the game:
![20201223230606_1](https://user-images.githubusercontent.com/10966059/103036959-8c293b00-4573-11eb-929a-5362e4446cb7.jpg)

And i have "Per pixel" alpha sorting enabled.

**Comment by taydan04** [comment]
> > @Scr34mik
> > Yeah i had this issue. But I have a solution for you. Here is version [aeb9e59](https://github.com/libretro/flycast/commit/aeb9e59beb572e856d0c0c199929d50a66d3f7b4) (Current commit) of the core that I built for windows x86_64 from source to get around it. Link will expire in a few days. Hmu if you don't get to it quick enough. Just unzip the dll file in to the cores folder within the RetroArch folder. :)
> > [flycast_libretro.zip](https://1drv.ms/u/s!Ap0VEg7UjYqTgo4SUL32AEm_x-IrtA?e=dG7pXg)
> > And if you don't trust the shared file (understandable) then there is a tutorial on the libretro site for building cores in windows [here](https://docs.libretro.com/development/retroarch/compilation/windows/).
> 
> Hi, thanks for the file, it'll be useful. 👍
> I was in no rush however, and could just wait until buildbot bug is resolved.
> 
> UPD: Thanks. it seems issue with save fiels files is resolved. However, now there are corrupted polygons in the game:
> ![20201223230606_1](https://user-images.githubusercontent.com/10966059/103036959-8c293b00-4573-11eb-929a-5362e4446cb7.jpg)
> 
> And i have "Per pixel" alpha sorting enabled.

Hmm, I haven't seen this before although I haven't tried this game. What video driver are you using in retroarch? (Under settings/drivers) I had issues in shenmue with per-pixel sorting when using the default gl driver. Switching to Vulkan fixed this for me though. Also are you using a BIOS file or HLE? I doubt that would cause it but could be worth trying the one or the other. 👍

**Comment by Scr34mik** [comment]
1. GL. Using Vulkan results in black screen and/or decreases performance. Previously GL worked fine with "Per-pixel" alpha sorting.
2. BIOS file.. I tried HLE as well, same issues.

**Comment by taydan04** [comment]
@Scr34mik 

Heya,

I just tracked down a GDI to try to reproduce the issue and I only get the issue with the gl video driver. The game runs nicely for me with the Vulkan driver. And thats on an Intel iGPU on my macbook air in bootcamp. No blackscreens or slowdowns i noticed in the brief play test I did trying a practice match on Babylon. I am playing at the standard 640x480 though, not sure if you are running a higher internal resolution or not and don't know how well the Vulkan renderer handles this. But as soon as i then played the practice match again with the gl driver most of the map had random glitching transparency issues.

I guess this could be a difference between the GPU's we are using why you have issues in Vulkan that I'm not seeing. Possibly a problem between the nvidia drivers and the vulkan renderer maybe. I'll give it a try tomorrow on my desktop that has a GTX1070 and see if Vulkan causes me problems.

But for now all I can tell you is that Vulkan makes this problem go away for me. If Vulkan is no good for you have you tried the other renderers like DirectX? I haven't tried them myself. But it definitely seems to be a problem between this version of the core and the gl renderer.

**Comment by Scr34mik** [comment]
@taydan04 
I'm running with 1280x960 internal. The game runs almost the same on both 640p and 960p in terms of performance, so i've chosen the latter.
Now you've mentioned DirectX. It has serious FPS dips.. especially DX12. And it doesn't depend on which internal resolution i set. Try playing a match on "**DM-Healpod**" or "**DM-Morpheus**" You'll see what i mean. GL perfroms better in these instances.

I think the best option for me at this point, would be to just find some older build of Flycast where everything worked, and never update it for a year or so.

**UPD:**
Found a way to restore previous core that has no issues with GL and VMUs. If u're one of unlucky ones like me, try my restored copy:
[flycast_libretro_b783996.zip](https://github.com/libretro/flycast/files/5743038/flycast_libretro_b783996.zip)
Should be "b783996" in window title when ure running a game.
